lib/shared/types.js

```diff
@@ -189,11 +189,32 @@ module.exports = {};
  * @property {number} warningCount Number of warnings for the result.
  * @property {number} fixableErrorCount Number of fixable errors for the result.
  * @property {number} fixableWarningCount Number of fixable warnings for the result.
+ * @property {Stats} Stats The performance statistics collected with the `stats` flag.
  * @property {string} [source] The source code of the file that was linted.
  * @property {string} [output] The source code of the file that was linted, with as many fixes applied as possible.
  * @property {DeprecatedRuleInfo[]} usedDeprecatedRules The list of used deprecated rules.
  */
 
+/**
+ * Performance statistics
+ * @typedef {Object} Stats
+ * @property {number} fixPasses The number of times ESLint has applied at least one fix after linting.
+ * @property {Times} times The times spent on (parsing, fixing, linting) a file.
+ */
+
+/**
+ * Performance Times for each ESLint pass
+ * @typedef {Times[]} Time passes
+ */
+
+/**
+ * @typedef {Object} Times
+ * @property {number} fix The time for the fix pass.
+ * @property {number} parse The time that is spent when parsing a file.
+ * @property {Object} rules Object of times for each rule.
```

Comment:
```suggestion
 * @property {Object<string, number>} rules Object of times for each rule.
```

Reply:
That may be fine although I believe `Record<...>` is more common.